src/commonMain/kotlin/app/cash/turbine/FlowTurbine.kt

```diff
@@ -261,6 +275,25 @@ private class ChannelBasedFlowTurbine<T>(
     return event.throwable
   }
 
+  override suspend fun expectMostRecentItem(): T {
+    var item: T? = null
+    while (true) {
+      try {
+        when (val event = expectEvent()) {
```

Comment:
I neglected to look closely at the implementation of this function and I don't think it's correct. If you have a Flow that emits every .5 seconds and are using an item timeout of 1 second this function will never return.

I think what we want is to make this function poll the channel directly for events which will mean a few things:
 1. It will not longer honor the timeout and no longer needs to be suspending.
 2. It will throw an AssertionError if no events have been received. You will not be able to _either_ wait or get the most recent, but I think that's fine since that indicates a bad test.

We probably also need one or two more tests to validate the change behaves as expected with situations like the .5s/1s thing.

If you want, I can take over the PR and do these changes. Or you can keep going if you want!

Reply:
I would like to continue working on that! I've made the changes and also found that with the previous implementation `expectMostRecentItem` would crash if the `Flow` could emit a null value. I've handled that by using a wrapper around the `Event.Item<T>`.